lib/utils/retry.go

```diff
@@ -83,6 +85,8 @@ type LinearConfig struct {
 	// AutoReset, if greater than zero, causes the linear retry to automatically
 	// reset after Max * AutoReset has elapsed since the last call to Incr.
 	AutoReset int64
+	// Clock to override clock in tests
+	Clock clockwork.Clock
```

Comment:
couldn't comment inline below: `func Inc` still uses `time.Now()` instead of `r.Clock.Now()`